renku/service/controllers/cache_list_projects.py

```diff
@@ -40,7 +40,11 @@ def context(self):
     def list_projects(self):
         """List locally cache projects."""
         projects = [project for project in self.cache.get_projects(self.user) if project.abs_path.exists()]
-        return {"projects": projects}
+        projects.sort(key=lambda p: p.created_at)
+
+        latest = {p.git_url: p for p in projects}
+
+        return {"projects": list(latest.values())}
```

Comment:
```suggestion

        return {
            "projects": [
                 max(g, key=lambda p: p.created_at) 
                     for _, g in itertools.groupby(projects, lambda p: p.git_url)
             ]
         }
```
Is maybe a bit more readable and doesn't rely on a side-effect of how dictionaries work in python.